src/client/action/watch.rs

```diff
@@ -0,0 +1,350 @@
+use std::{future::IntoFuture, time::Duration};
+
+use bson::{Bson, Document, Timestamp};
+use futures_util::FutureExt;
+
+use super::option_setters;
+use crate::{
+    change_stream::{
+        event::{ChangeStreamEvent, ResumeToken},
+        options::{ChangeStreamOptions, FullDocumentBeforeChangeType, FullDocumentType},
+        session::SessionChangeStream,
+        ChangeStream,
+    },
+    client::BoxFuture,
+    collation::Collation,
+    error::{ErrorKind, Result},
+    operation::AggregateTarget,
+    options::ReadConcern,
+    selection_criteria::SelectionCriteria,
+    Client,
+    ClientSession,
+    Collection,
+    Database,
+};
+
+impl Client {
+    /// Starts a new [`ChangeStream`] that receives events for all changes in the cluster. The
+    /// stream does not observe changes from system collections or the "config", "local" or
+    /// "admin" databases. Note that this method (`watch` on a cluster) is only supported in
+    /// MongoDB 4.0 or greater.
+    ///
+    /// See the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/) on change
+    /// streams.
+    ///
+    /// Change streams require either a "majority" read concern or no read
+    /// concern. Anything else will cause a server error.
+    ///
+    /// Note that using a `$project` stage to remove any of the `_id` `operationType` or `ns` fields
+    /// will cause an error. The driver requires these fields to support resumability. For
+    /// more information on resumability, see the documentation for
+    /// [`ChangeStream`](change_stream/struct.ChangeStream.html)
+    ///
+    /// If the pipeline alters the structure of the returned events, the parsed type will need to be
+    /// changed via [`ChangeStream::with_type`].
+    pub fn watch(&self) -> Watch {
+        Watch::new_cluster(&self)
+    }
+}
+
+impl Database {
```

Comment:
Since the impls for `Database` and `Collection` are just different `AggregateTargets`, I opted to just bundle all of them into this file.